src/components/Cluster/ClusterDetail/ClusterDetail.js

```diff
@@ -1,38 +1,57 @@
+import { push } from 'connected-react-router';
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
 import PropTypes from 'prop-types';
 import React from 'react';
 import { Breadcrumb } from 'react-breadcrumbs';
-import { connect } from 'react-redux';
+import { connect, useDispatch } from 'react-redux';
 import { Redirect, Route, Switch } from 'react-router-dom';
 import { selectCanClusterUpgrade } from 'selectors/clusterSelectors';
 import cmp from 'semver-compare';
+import { AppRoutes } from 'shared/constants/routes';
 import _ from 'underscore';
 
 import GettingStarted from '../../GettingStarted/GettingStarted';
 import ClusterDetailView from './ClusterDetailView';
 
-const ClusterDetail = props => (
-  <Breadcrumb
-    data={{
-      title: props.match.params.clusterId,
-      pathname: props.match.url,
-    }}
-  >
-    <Switch>
-      <Route
-        exact
-        path={`${props.match.path}`}
-        render={() => <ClusterDetailView {...props} />}
-      />
-
-      <Route
-        path={`${props.match.path}/getting-started/`}
-        render={() => <GettingStarted {...props} />}
-      />
-
-      <Redirect path={`${props.match.path}/*`} to={`${props.match.url}`} />
-    </Switch>
-  </Breadcrumb>
-);
+const ClusterDetail = props => {
+  const dispatch = useDispatch();
+
+  if (!props.cluster) {
+    new FlashMessage(
+      `Cluster <code>${props.clusterId}</code> doesn't exist.`,
+      messageType.INFO,
+      messageTTL.MEDIUM
+    );
+
+    dispatch(push(AppRoutes.Home));
+
+    return null;
+  }
```

Comment:
I think this should be inside an `useEffect` hook like this:

```js
useEffect(() => {
  if (!props.cluster) {
    new FlashMessage(
      `Cluster <code>${props.clusterId}</code> doesn't exist.`,
      messageType.INFO,
      messageTTL.MEDIUM
    );

    dispatch(push(AppRoutes.Home));
},[]) // empty dependencies to make it behave like componentDidMount
```

Reply:
`useEffect` runs after every render, and I want an early return ¿why wait to render when we can redirect before?

Reply:
Good point, with `useEffect` the conditional would be executed only once which saves the component from evaluating the if clause on every render.

Maybe checking this here is not the right way to go anyway?

Reply:
Maybe :-) but, ¿Why? And where would you place the check?

Reply:
After some thinking I couldn't find a better place ;D

My comment about `useEffect` still stands. When used with an empty array as dependencies, the function inside gets executed only once.

See https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect

Reply:
If I were to add a `useEffect` I would have to check before (after in the code, before in time) in render that the prop exist, otherwise it throws an error because the render goes first, so the condition will be evaluated in every render as well.

So my conclusion is that there is nothing we get from adding the `useEffect` ¿Or maybe am I missing something?

Reply:
oh yeah you're right. I didn't read the rest of the component 😮